### PR TITLE
fix(use-sort-by): sorting now ignores column ids that no longer exist

### PR DESCRIPTION
**Pull request by paolostyle**
I have a table that dynamically generates columns (there is a switch that transposes rows to columns and vice versa). However if I sort the existing table and then click that "switch" button, I'm getting error `Cannot read property 'sortType' of undefined`, because the column by which the content was sorted before no longer exists. This PR fixes this issue - if the column no longer exists, the sorting function for it always returns 0 (so the items stay in its original place). I don't remove them from `sortBy` state, because, as I said, my table has a switch, so if I go back to previous view it will keep the sorting that I set before switching. It works the same way with filters, so I think it's fine.

By the way, I'd recommend installing Prettier as dev dependency instead of relying on globally installed one (I suppose that's how it works right now) - I modified only small part of this file but it reformatted much more than that and I guess it's because of version mismatch - I didn't have Prettier installed globally and I had to install the latest one. But that's a topic for a separate PR most likely.

edit: It seems like in alpha.22 it doesn't work with filters as well... I can prepare a separate PR for that too but now I'm not even sure if you want to support this kind of behavior, though I think in any case the component should not completely fail...

### CODE REVIEW

**Comment by tannerlinsley** [comment]
I just installed prettier and did a global format. You should be able to clean up your diffs now by doing the same.

**Comment by tannerlinsley** [comment]
`yarn && yarn format` should do the trick

**Comment by paolostyle** [comment]
Thanks! Done.

**Comment by paolostyle** [comment]
Okay, I took a closer look at the issue with filters and it seems like it was just a mistake on your side, you were assigning value to a property of column before checking if column exists, and that column check was just below that. So I fixed that in this PR.

File: src/plugin-hooks/useSortBy.js [hunk 1]
```diff
@@ -217,31 +217,38 @@ function useMain(instance) {
...
+              sortTypes.alphanumeric
+
+            // Return the correct sortFn
+            return (a, b) =>
+              sortMethod(a.values[sort.id], b.values[sort.id], sort.desc)
+          } else {
+            // if a column with id from sortBy state doesn't exist,
+            // return sorting function that keeps items in its original place
+            return () => 0
+          }
```
Comment (tannerlinsley) [inline src/plugin-hooks/useSortBy.js]: Another way we could do this is filter the sortBy by column existence before the map. Then we don't need to fake anything at all. Thoughts?

Comment (paolostyle) [reply src/plugin-hooks/useSortBy.js]: Sounds good, it's probably more performant for very large lists if there's a few of those columns that don't exist. Will take care of that.

File: src/plugin-hooks/useSortBy.js [hunk 2]
```diff
@@ -209,42 +209,48 @@ function useMain(instance) {
...
-              sortTypes.alphanumeric
-
-            // Return the correct sortFn
-            return (a, b) =>
-              sortMethod(a.values[sort.id], b.values[sort.id], sort.desc)
-          } else {
-            // if a column with id from sortBy state doesn't exist,
-            // return sorting function that keeps items in its original place
-            return () => 0
+          if (!column) {
```
Comment (tannerlinsley) [inline src/plugin-hooks/useSortBy.js]: This is technically an impossible logical tree now ;)

Comment (paolostyle) [reply src/plugin-hooks/useSortBy.js]: That's true, but I guess I have some habits from TypeScript, you'd have to check column for null anyway. I can remove it, up to you.

Comment (tannerlinsley) [reply src/plugin-hooks/useSortBy.js]: No it's fine. The bundle size difference is negligible anyway.

**Pull request merged**